hcl2template/types.packer_config.go

```diff
@@ -123,7 +123,7 @@ func (cfg *PackerConfig) EvalContext(ctx BlockContext, variables map[string]cty.
 
 	iterID, ok := cfg.HCPVars["iterationID"]
 	if ok {
-		log.Printf("iterationID set: %q", iterID)
+		log.Printf("iterationID set: %q", iterID.AsString())
```

Comment:
Is this a log that we want to print? What's the value of it?

Reply:
This is a valid call out. I updated the PR to remove the log since it is repetitive and doesn't add info to the debug output. 

```
2023/08/07 15:31:03 [TRACE] a valid iteration for build was created with the Id 01H789Z1RP4206EJ5Q3V8SBYVV
2023/08/07 15:31:03 [TRACE] a valid iteration was retrieved with the id 01H789Z1RP4206EJ5Q3V8SBYVV
2023/08/07 15:31:03 [TRACE] registering build with iteration for "file.test".
2023/08/07 15:31:03 iterationID set: "01H789Z1RP4206EJ5Q3V8SBYVV"
2023/08/07 15:31:03 [INFO] Starting internal plugin packer-builder-file
2023/08/07 15:31:03 Starting plugin: /home/vagrant/Development/packer/bin/packer []string{"/home/vagrant/Development/packer/bin/packer", "plugin", "packer-builder-file"}
2023/08/07 15:31:03 Waiting for RPC address for: /home/vagrant/Development/packer/bin/packer
2023/08/07 15:31:03 packer-builder-file plugin: [INFO] Packer version: 1.10.0-dev [go1.20.3 linux amd64]
2023/08/07 15:31:03 packer-builder-file plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/08/07 15:31:03 packer-builder-file plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/vagrant/.packerconfig
2023/08/07 15:31:03 packer-builder-file plugin: [WARN] Config file doesn't exist: /home/vagrant/.packerconfig
2023/08/07 15:31:03 packer-builder-file plugin: [INFO] Setting cache directory: /home/vagrant/.cache/packer
2023/08/07 15:31:03 packer-builder-file plugin: args: []string{"packer-builder-file"}
2023/08/07 15:31:03 packer-builder-file plugin: Plugin address: unix /tmp/packer-plugin552367049
2023/08/07 15:31:03 packer-builder-file plugin: Waiting for connection...
2023/08/07 15:31:03 Received unix RPC address for /home/vagrant/Development/packer/bin/packer: addr is /tmp/packer-plugin552367049
2023/08/07 15:31:03 packer-builder-file plugin: Serving a plugin connection...
2023/08/07 15:31:03 iterationID set: "01H789Z1RP4206EJ5Q3V8SBYVV"
2023/08/07 15:31:03 iterationID set: "01H789Z1RP4206EJ5Q3V8SBYVV"
2023/08/07 15:31:03 iterationID set: "01H789Z1RP4206EJ5Q3V8SBYVV"
```

Reply:
Agreed with Moss on that, I think I added this as a debug print statement, but forgot to remove it when introducing this code, so we can definitely remove it.